theano/gof/opt.py

```diff
@@ -1261,10 +1294,82 @@ def tracks(self):
         return t
 
     def transform(self, node):
-        for opt in self.opts:
-            repl = opt.transform(node)
-            if repl:
-                return repl
+        if len(self.opts) == 0:
+            return
+
+        def apply_mult_opts(node, fgraph, multiple_opts=False, prev_repl=None):
+            opts = self.track_map[type(node.op)] + self.track_map[node.op] + self.track_map[None]
+            repl = prev_repl
+            new_repl = None
+            for opt in opts:
+                opt_start = time.time()
+                repl = opt.transform(node)
+                opt_finish = time.time()
+                if self.profile:
+                    self.time_opts[opt] += opt_start - opt_finish
+                    self.process_count[opt] += 1
+                if not repl:
+                    continue
+                else:
+                    if self.profile:
+                        self.node_created[opt] += len(graph.ops(fgraph.variables, repl))
+                        self.applied_true[opt] += 1
+                    if not multiple_opts or not repl[0].owner:
+                        return repl
+                    assert len(repl) == 1
+                    # Ensuring not the input of graph
+                    assert repl[0].owner
```

Comment:
remove this assert. It is useless. We have the condition on line 1317 that make sure it is always true.